maat-court-data-api/src/main/java/gov/uk/courtdata/eform/validator/TypeValidator.java

```diff
@@ -0,0 +1,19 @@
+package gov.uk.courtdata.eform.validator;
+
+import gov.uk.courtdata.exception.ValidationException;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+/**
+ * The responsibility of this class is to validate eForms types i.e. EFORMS_STAGING.TYPE
+ */
+@Component
+public class TypeValidator {
+
+    public void validate(String type) throws ValidationException {
+        if (StringUtils.isBlank(type)) {
+            String message = String.format("Expected non-null, non-empty, non-blank type but found [%s]", type);
+            throw new ValidationException(message);
+        }
+    }
+}
```

Comment:
This class is not required as Type is nullable filed

Reply:
see note on validation call comment. there to force the not null from requester of endpoint.

Reply:
been removed now from repo